src/components/SidePanel/SidePanel.jsx

```diff
@@ -0,0 +1,178 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useState } from "react";
+import {
+  ContentContainer,
+  Button,
+  ContentName,
+  SelectStyle,
+  Icon,
+} from "components/SidePanel/styles";
+import PropTypes from "prop-types";
+import SearchAllAvailabe from "components/SearchAllAvailabe/SearchAllAvailabe";
+import GenereInput from "components/GenereInput/GenereInput";
+import axios from "axios";
+import SortInput from "components/SortInput/SortInput";
+import ReleaseDate from "components/ReleaseDate/ReleaseDate";
+import Keyword from "components/Keyword/Keyword";
+import arrow from "assets/arrow.svg";
+import { Paragraph } from "components/shared-styled-comp/shared-styled-comp";
+/**
+ * Filter and sort component for movies.
+ * @param {object} props - The properties for the component.
+ * @param {function} props.getMovies - Function to retrieve movies.
+ * @param {function} props.setSelectedGenres - Function to set selected genres.
+ * @param {Array} props.selectedGenres - Array of selected genres.
+ * @param {function} props.setAvailabilities - Function to set availabilities.
+ * @param {Array} props.availabilities - Array of availabilities.
+ * @param {function} props.setRelease - Function to set release.
+ * @param {Array} props.release - Array of release types.
+ * @returns {JSX.Element} The SidePanel component.
+ */
+
+export default function SidePanel({
+  getMovies,
+  setSelectedGenres,
+  selectedGenres,
+  setAvailabilities,
+  availabilities,
+  setRelease,
+  release,
+  setSelectedKeyword,
+  selectedKeyword,
+}) {
+  const [isActiveFilter, setIsActiveFilter] = useState(false);
+  const [lang, setLang] = useState([]);
+  const [country, setCountry] = useState([]);
+  const [genres, setGenres] = useState([]);
+
+  /**
+   * Fetches movie genres from the API and sets them using the provided callback.
+   */
+  const getGenres = () => {
+    const apiKey = "4db3b4ee5893cead9657d41699ec4c26";
+    const apiUrl = `https://api.themoviedb.org/3/genre/movie/list?language=en&api_key=${apiKey}`;
+    axios
+      .get(apiUrl)
+      .then((response) => {
+        setGenres(response.data.genres);
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  };
+
+  /**
+   * Fetches available languages from the API and sets them using the provided callback.
+   */
+  const getLanguages = () => {
+    const apiKey = "4db3b4ee5893cead9657d41699ec4c26";
+    const apiUrl = `https://api.themoviedb.org/3/configuration/languages?api_key=${apiKey}`;
+    axios.get(apiUrl).then((response) => {
+      setLang(response.data);
+    });
+  };
+
+  /**
+   * Fetches available countries from the API and sets them using the provided callback.
+   */
+  const getCountry = () => {
+    const apiKey = "4db3b4ee5893cead9657d41699ec4c26";
+    const apiUrl = `https://api.themoviedb.org/3/configuration/countries?language=en-US&api_key=${apiKey}`;
+    axios.get(apiUrl).then((response) => {
```

Comment:
I think it's a great idea to create a utility function for making Axios requests with built-in error validation. This would make error-handling logic consistent for all requests.